packages/react/src/ReactStartTransition.js

```diff
@@ -6,17 +6,56 @@
  *
  * @flow
  */
+import type {StartTransitionOptions} from 'shared/ReactTypes';
 
 import ReactCurrentBatchConfig from './ReactCurrentBatchConfig';
-import {warnOnSubscriptionInsideStartTransition} from 'shared/ReactFeatureFlags';
+import ReactCurrentHostConfig from './ReactCurrentHostConfig';
+import {
+  warnOnSubscriptionInsideStartTransition,
+  enableTransitionTracing,
+} from 'shared/ReactFeatureFlags';
 
-export function startTransition(scope: () => void) {
+// Intentionally not named imports because Rollup would use dynamic dispatch for
+// CommonJS interop named imports.
+import * as Scheduler from 'scheduler';
+
+const {unstable_now: now} = Scheduler;
+
+export function startTransition(
+  scope: () => void,
+  options?: StartTransitionOptions,
+) {
   const prevTransition = ReactCurrentBatchConfig.transition;
   ReactCurrentBatchConfig.transition = 1;
+
+  let prevTransitionInfo = null;
+  if (enableTransitionTracing) {
+    prevTransitionInfo = ReactCurrentBatchConfig.transitionInfo;
+    if (options !== undefined && options.name !== undefined) {
+      const hostConfig = ReactCurrentHostConfig.current;
```

Comment:
Dynamically swapping out the host config isn't going to work. The host config is static — it gets resolved by the compiler (Rollup in this case). You can't change it while the application is running. Or rather, you can, but only by using dynamic dispatch, which we don't want to do.

(By the way, you were asking what "dynamic" versus "static" dispatch means the other day. What you're doing here is an example of dynamic dispatch, where the implementation of a function can only be determined at runtime.)

What you can do instead is lazily assign the `startTime` inside `scheduleUpdateOnFiber`. Something like:

```js
// Inside react-reconciler/src/ReactFiberWorkLoop
import {getCurrentEventTime} from './ReactFiberHostConfig';

// scheduleUpdateOnFiber
const transition = ReactCurrentBatchConfig.transitionInfo;
if (transition.startTime = -1) {
  transition.startTime = getCurrentEventStartTime();
}
```